libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -521,7 +522,7 @@ void AP_Baro::update(void)
             // update altitude calculation
             float ground_pressure = sensors[i].ground_pressure;
             if (!is_positive(ground_pressure) || isnan(ground_pressure) || isinf(ground_pressure)) {
-                sensors[i].ground_pressure = sensors[i].pressure;
```

Comment:
From what I can see, the only time this line will normally be reached is if the sensor malfunctions or gives a bad reading during boot, when calibrate() is called. So we were trusting a sensor that has given bad data once to give good data the next time.

Also, there is no check that we are actually on the ground when we are setting the ground_pressure here. So if we were to set the ground pressure to the current pressure at altitude (say the parameter is edited to a negative number accidentally), then the baro thinks it is on the ground again mid-flight (and the ekf gets a large step-input).

Either way it's not 100% right in my view. I thought it could use a second look, and this is why I made it a separate commit at the end.

To answer your question, atmospheric/altitude variations only offset Sub's altitude on the order of decimeters at most.